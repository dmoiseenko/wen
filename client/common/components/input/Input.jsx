import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Label from './InputLabel';
import Icon from './InputIcon';
import InputError from './InputError';


export default function Input({
  meta,
  input,
  type,
  label,
  placeholder,
  inputClassName,
  iconClassName
}) {
  const error = meta.touched && !!meta.error;

  const className = classnames('input', inputClassName, {
    'is-danger': error
  });

  return (
    <div className="field" id={input.name}>
      <Label label={label} />
      <p className="control has-icon has-icon-right">
        <input
          {...input}
          className={className}
          type={type}
          placeholder={placeholder}
        />
        <Icon
          iconClassName={iconClassName}
          error={error}
        />
      </p>
      <InputError error={meta.error} />
    </div>
  );
}

Input.propTypes = {
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired,
  inputClassName: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  iconClassName: PropTypes.string.isRequired
};

Input.defaultProps = {
  label: '',
  inputClassName: '',
  iconClassName: ''
};
