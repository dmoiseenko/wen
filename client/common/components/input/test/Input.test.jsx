import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../Input';


it('should render', () => {
  const props = {
    meta: {
      touched: false,
    },
    input: {
      name: 'inputName',
    },
    type: 'inputType',
    label: 'inputLabel',
    placeholder: 'inputPlaceholder',
    inputClassName: 'inputClassName',
    iconClassName: 'iconClassName'
  };

  const tree = renderer.create(
    <Input {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should render error', () => {
  const props = {
    meta: {
      touched: true,
      error: 'error'
    },
    input: {
      name: 'inputName',
    },
    type: 'inputType',
    label: 'inputLabel',
    placeholder: 'inputPlaceholder',
    inputClassName: 'inputClassName',
    iconClassName: 'iconClassName'
  };

  const tree = renderer.create(
    <Input {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
