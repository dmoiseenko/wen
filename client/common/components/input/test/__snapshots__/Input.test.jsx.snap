// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="field"
  id="inputName"
>
  <label
    className="label"
  >
    inputLabel
  </label>
  <p
    className="control has-icon has-icon-right"
  >
    <input
      className="input inputClassName"
      name="inputName"
      placeholder="inputPlaceholder"
      type="inputType"
    />
    <span
      className="icon"
    >
      <i
        className="iconClassName"
      />
    </span>
  </p>
</div>
`;

exports[`should render error 1`] = `
<div
  className="field"
  id="inputName"
>
  <label
    className="label"
  >
    inputLabel
  </label>
  <p
    className="control has-icon has-icon-right"
  >
    <input
      className="input inputClassName is-danger"
      name="inputName"
      placeholder="inputPlaceholder"
      type="inputType"
    />
    <span
      className="icon"
    >
      <i
        className="fa fa-warning"
      />
    </span>
  </p>
  <p
    className="help is-danger"
  >
    error
  </p>
</div>
`;
